### ISSUE TITLE
Error when running model.py, something to do with RandomRotation?

### ISSUE THREAD

**Issue by Tarnekar** [QUVA-Lab/e2cnn#1]
When running model.ipynb, I get the error shown below when training, in box [9]:

When commenting out the RandomRotation in the train_transform function, there are no errors and I able to run the model without errors. 

Any idea what could be wrong?

And thanks a lot for this really nice implementation and tutorial!

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-be8ef5778d6d> in <module>
      2 for epoch in range(51):
      3     model.train()
----> 4     for i, (x, t) in enumerate(train_loader):
      5         if i > 20:
      6             break

~/anaconda3/lib/python3.7/site-packages/torch/utils/data/dataloader.py in __next__(self)
    343 
    344     def __next__(self):
--> 345         data = self._next_data()
    346         self._num_yielded += 1
    347         if self._dataset_kind == _DatasetKind.Iterable and \

~/anaconda3/lib/python3.7/site-packages/torch/utils/data/dataloader.py in _next_data(self)
    383     def _next_data(self):
    384         index = self._next_index()  # may raise StopIteration
--> 385         data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
    386         if self._pin_memory:
    387             data = _utils.pin_memory.pin_memory(data)

~/anaconda3/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py in fetch(self, possibly_batched_index)
     42     def fetch(self, possibly_batched_index):
     43         if self.auto_collation:
---> 44             data = [self.dataset[idx] for idx in possibly_batched_index]
     45         else:
     46             data = self.dataset[possibly_batched_index]

~/anaconda3/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py in <listcomp>(.0)
     42     def fetch(self, possibly_batched_index):
     43         if self.auto_collation:
---> 44             data = [self.dataset[idx] for idx in possibly_batched_index]
     45         else:
     46             data = self.dataset[possibly_batched_index]

<ipython-input-5-fab3333837e6> in __getitem__(self, index)
     21         image = Image.fromarray(image)
     22         if self.transform is not None:
---> 23             image = self.transform(image)
     24         return image, label
     25 

~/anaconda3/lib/python3.7/site-packages/torchvision/transforms/transforms.py in __call__(self, img)
     68     def __call__(self, img):
     69         for t in self.transforms:
---> 70             img = t(img)
     71         return img
     72 

~/anaconda3/lib/python3.7/site-packages/torchvision/transforms/transforms.py in __call__(self, img)
   1001         angle = self.get_params(self.degrees)
   1002 
-> 1003         return F.rotate(img, angle, self.resample, self.expand, self.center, self.fill)
   1004 
   1005     def __repr__(self):

~/anaconda3/lib/python3.7/site-packages/torchvision/transforms/functional.py in rotate(img, angle, resample, expand, center, fill)
    727         fill = tuple([fill] * 3)
    728 
--> 729     return img.rotate(angle, resample, expand, center, fillcolor=fill)
    730 
    731 

~/anaconda3/lib/python3.7/site-packages/PIL/Image.py in rotate(self, angle, resample, expand, center, translate, fillcolor)
   1913 
   1914         return self.transform((w, h), AFFINE, matrix, resample,
-> 1915                               fillcolor=fillcolor)
   1916 
   1917     def save(self, fp, format=None, **params):

~/anaconda3/lib/python3.7/site-packages/PIL/Image.py in transform(self, size, method, data, resample, fill, fillcolor)
   2203             raise ValueError("missing method data")
   2204 
-> 2205         im = new(self.mode, size, fillcolor)
   2206         if method == MESH:
   2207             # list of quads

~/anaconda3/lib/python3.7/site-packages/PIL/Image.py in new(mode, size, color)
   2373         color = ImageColor.getcolor(color, mode)
   2374 
-> 2375     return Image()._new(core.fill(mode, size, color))
   2376 
   2377 

TypeError: must be real number, not tuple

**Comment by Gabri95** [comment]
Hi @Tarnekar! I am happy you liked our work!

Unfortunately, I have never had this problem.
It seems its source is in Pytorch's code where PIL functionalities are used.
Is it possible that you have an old version of PIL which is not fully compatible with your current version of torchvision?

Could you give some more details about your Python environment?

**Comment by Tarnekar** [comment]
Thanks for your answer and support. I tried updating PIL from 5.4.1 to 7.0.0, and executing the code inside a virtual environment but still the same problem occurs.

Here are the package version (Python 3.7 is used)

(e2cnn_2) karl@IT-LL-karbe297:~$ pip list
Package            Version
------------------ -------
attrs              19.3.0 
backcall           0.1.0  
bleach             3.1.0  
decorator          4.4.1  
defusedxml         0.6.0  
e2cnn              0.1    
entrypoints        0.3    
importlib-metadata 1.4.0  
ipykernel          5.1.3  
ipython            7.11.1 
ipython-genutils   0.2.0  
jedi               0.15.2 
Jinja2             2.10.3 
json5              0.8.5  
jsonschema         3.2.0  
jupyter-client     5.3.4  
jupyter-core       4.6.1  
jupyterlab         1.2.5  
jupyterlab-server  1.0.6  
MarkupSafe         1.1.1  
mistune            0.8.4  
more-itertools     8.1.0  
nbconvert          5.6.1  
nbformat           5.0.3  
notebook           6.0.2  
numpy              1.18.1 
pandocfilters      1.4.2  
parso              0.5.2  
pexpect            4.7.0  
pickleshare        0.7.5  
Pillow             7.0.0  
pip                20.0.1 
pkg-resources      0.0.0  
prometheus-client  0.7.1  
prompt-toolkit     3.0.2  
ptyprocess         0.6.0  
Pygments           2.5.2  
pyrsistent         0.15.7 
python-dateutil    2.8.1  
pyzmq              18.1.1 
scipy              1.4.1  
Send2Trash         1.5.0  
setuptools         45.1.0 
six                1.14.0 
terminado          0.8.3  
testpath           0.4.4  
torch              1.4.0  
torchvision        0.5.0  
tornado            6.0.3  
traitlets          4.3.3  
wcwidth            0.1.8  
webencodings       0.5.1  
wheel              0.33.6 
zipp               2.0.0

**Comment by Gabri95** [comment]
I was indeed able to reproduce your problem.
Using `torch` 1.3 and `torchvision` 0.4 (and `pillow` 6.2.1), the following code runs.
However, I get your error when I update to the newest (recently released) versions of `torch` (1.4) and `torchvision` (0.5).

```
import torch
from torch.utils.data import DataLoader
from torchvision.transforms import RandomRotation
from torchvision.transforms import ToTensor
from torchvision.transforms import Compose
from torchvision.datasets import MNIST
from PIL import Image


transform = Compose([
    RandomRotation(180., resample=Image.BILINEAR, expand=False),
    ToTensor(),
])

mnist = MNIST("./", train=True, transform=transform, download=True)
train_loader = DataLoader(mnist, batch_size=1)


for x, t in enumerate(train_loader):
    print("hello world")
    break
```

For this reason, I think this is a problem with the new version of Pytorch.
I would suggest you to temporarily downgrade it to 1.3 if you want to use `transforms.RandomRotation`.
You can probably find more support on Pytorch's forum.

Hope this helps!

**Comment by Tarnekar** [comment]
That was indeed the problem; using torch 1.3, torchvision 0.4 and pillow 6.2.1 and all the code runs. Although there is a warning that torchvision 0.4.0 has a requirement of torch==1.2.0.

Thanks!
